library/weaviate.py

```diff
@@ -1,120 +1,157 @@
-        ) as batch:
-            for row in values:
-                if row.page_content is not None and row.page_content != "":
-
-                    print("Upserting " + str(row))
-                    question_object = {
-                        "text": row.page_content,
+            print("Schema already exists")
+
+    def upsertChunkedText(self, obj, key: WeaviateSchemas, metadataKey: WeaviateSchemas, splitOn: str) -> bool:
+        text = obj[splitOn]
+        split_text = self.split(text)
+        collection = self.collection(key)
+        metaCollection = self.collection(metadataKey)
+        with metaCollection.batch.dynamic() as batch:
+            batch.add_object(
+                    obj,
+                    uuid = w.util.generate_uuid5(obj)
+            )
+
+        with collection.batch.dynamic() as batch:
+            for value in split_text:
+                row = {
+                        "text": value.page_content,
                     }
-                    batch.add_data_object(
-                        question_object,
-                        class_name = class_obj['class'],
-                        uuid = w.util.generate_uuid5(question_object)
-                    )
-                else:
-                    print("Skipping row with no content")
-
+                batch.add_object(
+                    row,
+                    uuid = w.util.generate_uuid5(row)
+            )
         return True
     
-    def count(self) -> object:
-        return self.client.query.aggregate(class_obj['class']).with_meta_count().do()
+    def count(self, key: WeaviateSchemas) -> object:
+        collection = self.collection(key)
+        return collection.aggregate.over_all()
 
     def split(self, text:str) -> list:
-
-
         text_splitter = lang_splitter.CharacterTextSplitter(
             separator="\n\n",
             chunk_size=1000,
             chunk_overlap=200,
             length_function=len,
             is_separator_regex=False,
-)
+        )
         # text_splitter = lang_splitter.SemanticChunker(GPT4AllEmbeddings())
         return text_splitter.create_documents([text])
     
-    def search(self, query:str) -> list:
-
-        # items = self.client.collections.get(class_obj['class'])
-        # response = items.query.near_text(
-        #     query=query,
-        #     limit=5,
-        #     # target_vector="title_country",  # Specify the target vector for named vector collections
-        #     return_metadata=wvc.query.MetadataQuery(distance=True)
-        # )
-        print("Querying for " + query + " in " + class_obj['class'] + "...")
-        response = (
-            self.client.query.get(class_obj['class'], ["text"])
-            .with_near_text({
-                "concepts": [query]
-            })
-            .with_limit(2)
-            .with_additional(["distance"])
-            .do()
+    def search(self, query:str, key: WeaviateSchemas, limit: int = 5) -> list:
```

Comment:
This is all now using the v4 API from weaviate as opposed to the v3. I had to change a bunch of things and it didn't make sense to keep using the deprecated syntax